x-pack/plugins/security_solution/public/timelines/components/timeline/query_bar/eql/index.tsx

```diff
@@ -0,0 +1,200 @@
+import * as i18n from '../translations';
+import { getEqlOptions } from './selectors';
+
+interface TimelineEqlQueryBar {
+  index: string[];
+  eqlQueryBar: FieldValueQueryBar;
+}
+
+const defaultValues = {
+  index: [],
+  eqlQueryBar: {
+    query: { query: '', language: 'eql' },
+    filters: [],
+    saved_id: undefined,
+  },
+};
+
+const schema: FormSchema<TimelineEqlQueryBar> = {
+  index: {
+    fieldsToValidateOnChange: ['index', 'eqlQueryBar'],
+    validations: [],
+  },
+  eqlQueryBar: {
+    validations: [
+      {
+        validator: debounceAsync(eqlValidator, 300),
+      },
+    ],
+  },
+};
+
+const HiddenUseField = styled(UseField)`
+  display: none;
+`;
+
+export const EqlQueryBarTimeline = memo(
+  ({ expression, timelineId }: { expression: string; timelineId: string }) => {
+    const dispatch = useDispatch();
+    const [isQueryBarValid, setIsQueryBarValid] = useState(false);
+    const [isQueryBarValidating, setIsQueryBarValidating] = useState(false);
+    const getOptionsSelected = useMemo(() => getEqlOptions(), []);
+    const optionsSelected = useDeepEqualSelector((state) => getOptionsSelected(state, timelineId));
+
+    const { loading: indexPatternsLoading, indexPattern, selectedPatterns } = useSourcererScope(
+      SourcererScopeName.timeline
+    );
+
+    const initialState = {
+      ...defaultValues,
+      index: selectedPatterns.sort(),
+      eqlQueryBar: {
+        ...defaultValues.eqlQueryBar,
+        query: { query: expression, language: 'eql' },
+      },
+    };
+
+    const { form } = useForm<TimelineEqlQueryBar>({
+      defaultValue: initialState,
+      options: { stripEmptyFields: false },
+      schema,
+    });
+    const { getFields } = form;
+
+    const applyLanguageFilterQuery = useCallback(
+      (language: string) =>
+        dispatch(
+          timelineActions.applyKqlFilterQuery({
+            id: timelineId,
+            filterQuery: {
+              kuery: {
+                kind: language as KueryFilterQueryKind,
+                expression: '',
+              },
+              serializedQuery: '',
+            },
+          })
+        ),
+      [dispatch, timelineId]
+    );
+
+    const onOptionsChange = useCallback(
+      (field: FieldsEqlOptions, value: string | null) =>
+        dispatch(
+          timelineActions.updateEqlOptions({
+            id: timelineId,
+            field,
+            value,
+          })
+        ),
+      [dispatch, timelineId]
+    );
+
+    const [{ eqlQueryBar: formEqlQueryBar }] = useFormData<TimelineEqlQueryBar>({
+      form,
+      watch: ['eqlQueryBar'],
+    });
+
+    const optionsData = useMemo(
+      () => ({
+        keywordFields: indexPattern.fields
+          .filter((f) => f.esTypes?.includes('keyword'))
+          .map((f) => ({ label: f.name })),
+        dateFields: indexPattern.fields
+          .filter((f) => f.type === 'date')
+          .map((f) => ({ label: f.name })),
+        allFields: indexPattern.fields
```

Comment:
Consider changing `allFields` to `nonDateFields`, because `dateFields` above it only _includes_ fields of type `date` via:

```
.filter((f) => f.type === 'date')
```

and `allFields` is really the "opposite" of the above, as it _excludes_ date fields via:

```
.filter((f) => f.type !== 'date')
```

on the next line below.